### PR TITLE
My solution to task stream practice

### PR DESCRIPTION
**Pull request by Swyatik19**

### CODE REVIEW

File: src/main/java/practice/StreamPractice.java [hunk 1]
```diff
@@ -14,7 +18,13 @@ public class StreamPractice {
...
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMap(s -> Stream.of(s.split(",")))
+                .mapToInt(Integer::valueOf)
+                .filter(n -> n % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException(
+                        "Can't get min value from list : Here is our inputmvn " + numbers));
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:27]:
```suggestion
                        "Can't get min value from list : " + numbers));
```

File: src/main/java/practice/StreamPractice.java [hunk 2]
```diff
@@ -35,7 +50,11 @@ public Double getOddNumsAverage(List<Integer> numbers) {
      * Example: select men who can be recruited to army (from 18 to 27 years old inclusively).
      */
     public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge
+                        && p.getAge() <= toAge
+                        && p.getSex().equals(Person.Sex.MAN))
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:56]: [checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-core/java-eight-part-two/stream-api#pay-attention-to-what-is-a-better-way-to-compare-enum-values-equals-vs--)
Pay attention to what is a better way to compare Enum values: equals() vs == ?

**Review by Rommelua — CHANGES_REQUESTED** [review]
Before you create PR with your homework, you need to go through the checklist under the task and correct all the points described there. The mentor will not check the work until the checklist points are corrected

File: src/main/java/practice/StreamPractice.java (hunk 1 above)
Comment (Swyatik19) [reply src/main/java/practice/StreamPractice.java:27]: I did not understand what is going on, what exactly to check?

File: src/main/java/practice/StreamPractice.java (hunk 2 above)
Comment (Rommelua) [reply src/main/java/practice/StreamPractice.java:56]: not fixed

**Review by Rommelua — CHANGES_REQUESTED** [review]
If you do not understand any mentor's comment - welcome to Slack)

File: src/main/java/practice/StreamPractice.java (hunk 2 above)
Comment (Swyatik19) [reply src/main/java/practice/StreamPractice.java:56]: My apologies, I have already fixed everything, I just accidentally started the wrong changes

Comment (Rommelua) [reply src/main/java/practice/StreamPractice.java:56]: But it is still not fixed...

File: src/main/java/practice/StreamPractice.java [hunk 3]
```diff
@@ -50,7 +69,11 @@ public List<Person> selectMenByAge(List<Person> peopleList, int fromAge, int toA
      */
     public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
                                           int maleToAge, List<Person> peopleList) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getAge() >= fromAge
+                        && (p.getAge() <= femaleToAge && p.getSex().equals(Person.Sex.WOMAN)
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:74]: and here

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/practice/StreamPractice.java [hunk 4]
```diff
@@ -59,7 +82,11 @@ public List<Person> getWorkablePeople(int fromAge, int femaleToAge,
      * return the names of all cats whose owners are women from `femaleAge` years old inclusively.
      */
     public List<String> getCatsNames(List<Person> peopleList, int femaleAge) {
-        return Collections.emptyList();
+        return peopleList.stream()
+                .filter(p -> p.getSex().equals(Person.Sex.WOMAN) && p.getAge() >= femaleAge)
```
Comment (Rommelua) [inline src/main/java/practice/StreamPractice.java:86]: еще чуть-чуть...))))

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/practice/StreamPractice.java (hunk 4 above)
Comment (Swyatik19) [reply src/main/java/practice/StreamPractice.java]: Sorry for my carelessness

**Review by aaananas — APPROVED** [review]

**Pull request closed without merging**